code/game/objects/items/potions.dm

```diff
@@ -193,4 +203,45 @@
 		while(top_level)
 			top_level = top_level.transmogrify()
 		if(T2)
-			playsound(T2, 'sound/effects/phasein.ogg', 50, 1)
\ No newline at end of file
+			playsound(T2, 'sound/effects/phasein.ogg', 50, 1)
+
+/obj/item/potion/toxin
+	name = "draught of toad"
+	desc = "Become immune to toxins."
+	icon_state = "green_emflask"
+
+/obj/item/potion/toxin/imbibe_effect(mob/user)
+	user.tox_damage_modifier = 0
+
+/obj/item/potion/zombie
+	name = "phial of exanimis"
+	desc = "Turn the dead into undead."
+	icon_state = "necro_flask2"
+
+/obj/item/potion/zombie/imbibe_check(mob/user)
+	if(ishuman(user))
+		return TRUE
+	return FALSE
+
+/obj/item/potion/zombie/imbibe_effect(mob/living/carbon/human/user)
+	user.become_zombie_after_death = TRUE
+	spawn(20)
+		to_chat(user, "<span class='notice'>Nothing seems to happen.</span>")
+
+/obj/item/potion/zombie/impact_atom(atom/target)
+	var/client/thrower = directory[ckey(fingerprintslast)]
```

Comment:
Are you sure this is reliable? Why the `ckey` call? There absolutely has to be a better way to get this.